File: rest/server/Server/Models/DTOs/Contribution/Responses/GetContributionDetailDto.cs
namespace Server.Models.DTOs.Contribution.Responses
{
    public class GetContributionDetailDto
    {
        public Guid ProjectID { get; set; }
        public Guid UserID { get; set; }
        public string Category { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
        public string Date { get; set; }
        public string Status { get; set; }
        public CreatorDetailDto Creator { get; set; }
        public List<VoteDetailDto> Votes { get; set; } = new();
        public List<CommentDetailDto> Comments { get; set; } = new();
    }

    public class CreatorDetailDto
    {
        public string Name { get; set; }
        public byte[] ProfilPicture { get; set; }
    }

    public class VoteDetailDto
    {
        public Guid VoteID { get; set; }
        public Guid UserID { get; set; }
    }

    public class CommentDetailDto
    {
        public string Text { get; set; }
        public string Date { get; set; }
        public CreatorDetailDto Creator { get; set; }
    }
}

File: graphql/client/src/app/models/dtos/contribution/responses/get-contribution-detail-dto/get-contribution-detail-dto.ts
export class GetContributionDetailDto {
    constructor(
        public projectID: string = '',
        public userID: string = '',
        public category: string = '',
        public title: string = '',
        public text: string = '',
        public date: string = '',
        public status: string = '',
        public creator: CreatorDto = new CreatorDto('', ''),
        public votes: VoteDto[] = [],
        public comments: CommentDto[] = [],
        ) {}
}

export class CreatorDto {
    constructor(
      public name: string,
      public profilPicture: string,
    ) {}
  }

export class VoteDto {
    constructor(
      public voteID: string,
      public userID: string,
    ) {}
  }
  
  export class CommentDto {
    constructor(
      public text: string,
      public date: string,
      public creator: CreatorDto,
    ) {}
  }
